tests/e2e/wdio.conf.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
+/*
+ * Usage:
+ *   wdio tests/wdio.conf.js [--target=firefox,chrome] [--debug] [--clean]
+ *
+ * Options:
+ *   --target: comma separated list of browsers to run the tests on (default: firefox,chrome)
+ *   --debug: run the tests in debug mode (default: false)
+ *   --clean: clean the build artifacts before running the tests (default: false)
+ */
+
+import path from 'node:path';
+import url from 'node:url';
+import { readFileSync, cpSync, existsSync, rmSync } from 'node:fs';
+import { createServer } from 'node:http';
+import { execSync } from 'node:child_process';
+
+export const WEB_EXT_PATH = path.join(process.cwd(), 'web-ext-artifacts');
+
+export const FIREFOX_PATH = path.join(WEB_EXT_PATH, 'ghostery-firefox.zip');
+export const CHROME_PATH = path.join(WEB_EXT_PATH, 'ghostery-chromium');
+
+const PAGE_PORT = 6789;
+export const PAGE_URL = `http://page.localhost:${PAGE_PORT}/`;
+
+// Generate arguments from command line
+export const argv = process.argv.slice(2).reduce(
+  (acc, arg) => {
+    if (arg.startsWith('--')) {
+      if (arg.includes('=')) {
+        const [key, value] = arg.slice(2).split('=');
+        acc[key] = value;
+      } else {
+        acc[arg.slice(2)] = true;
+      }
+    }
+    return acc;
+  },
+  { target: ['firefox', 'chrome'], debug: false, clean: false },
+);
+
+const __dirname = url.fileURLToPath(new URL('.', import.meta.url));
+export function setupTestPage() {
+  const file = readFileSync(path.join(__dirname, 'page.html'), 'utf8');
+  const server = createServer((req, res) => {
+    res.writeHead(200, { 'Content-Type': 'text/html' });
+    res.end(file);
+  });
+
+  // starts a simple http server locally on port 6789
+  server.listen(PAGE_PORT, '127.0.0.1', () => {
```

Comment:
we should close the server after the test

Reply:
This server is supposed to run during the whole testing process. It is set up only once and cleared out by the Nodejs itself. Look a the nodejs.org example from the home page - it does not closes itself too ;)